File: src/services/azienda.ts
import { invoke } from '@tauri-apps/api/core';

export interface Azienda {
  id: string;
  nome_centro: string;
  indirizzo?: string;
  citta?: string;
  cap?: string;
  provincia?: string;
  telefono?: string;
  email?: string;
  piva?: string;
  orario_apertura: string;
  orario_chiusura: string;
  slot_durata_minuti: number;
  giorni_lavorativi: string; // JSON string
  created_at: string;
  updated_at: string;
}

export interface UpdateAziendaInput {
  nome_centro: string;
  indirizzo?: string;
  citta?: string;
  cap?: string;
  provincia?: string;
  telefono?: string;
  email?: string;
  piva?: string;
  orario_apertura: string;
  orario_chiusura: string;
  slot_durata_minuti: number;
  giorni_lavorativi: string;
}

export interface OrarioCentro {
  id: string;
  giorno: number; // 0=Lun, 6=Dom
  attivo: boolean;
  mattina_inizio: string | null;
  mattina_fine: string | null;
  pomeriggio_inizio: string | null;
  pomeriggio_fine: string | null;
}

export interface UpdateOrarioCentroInput {
  giorno: number;
  attivo: boolean;
  mattina_inizio: string | null;
  mattina_fine: string | null;
  pomeriggio_inizio: string | null;
  pomeriggio_fine: string | null;
}

export const GIORNI_SETTIMANA = ['Lunedì', 'Martedì', 'Mercoledì', 'Giovedì', 'Venerdì', 'Sabato', 'Domenica'];

export const aziendaService = {
  async getAzienda(): Promise<Azienda> {
    return await invoke('get_azienda');
  },

  async updateAzienda(data: UpdateAziendaInput): Promise<Azienda> {
    return await invoke('update_azienda', { data });
  },

  async getOrariCentro(): Promise<OrarioCentro[]> {
    return await invoke('get_orari_centro');
  },

  async updateOrariCentro(orari: UpdateOrarioCentroInput[]): Promise<OrarioCentro[]> {
    return await invoke('update_orari_centro', { orari });
  },
};

File: src-tauri/src/models/azienda.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow)]
pub struct Azienda {
    pub id: String,
    pub nome_centro: String,
    pub indirizzo: Option<String>,
    pub citta: Option<String>,
    pub cap: Option<String>,
    pub provincia: Option<String>,
    pub telefono: Option<String>,
    pub email: Option<String>,
    pub piva: Option<String>,
    #[serde(skip_serializing)]
    pub logo: Option<Vec<u8>>,
    pub orario_apertura: String,
    pub orario_chiusura: String,
    pub slot_durata_minuti: i64,
    pub giorni_lavorativi: String, // JSON array
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateAziendaInput {
    pub nome_centro: String,
    pub indirizzo: Option<String>,
    pub citta: Option<String>,
    pub cap: Option<String>,
    pub provincia: Option<String>,
    pub telefono: Option<String>,
    pub email: Option<String>,
    pub piva: Option<String>,
    pub orario_apertura: String,
    pub orario_chiusura: String,
    pub slot_durata_minuti: i64,
    pub giorni_lavorativi: String,
}

#[derive(Debug, Serialize, Deserialize, sqlx::FromRow, Clone)]
pub struct OrarioCentro {
    pub id: String,
    pub giorno: i64,
    pub attivo: bool,
    pub mattina_inizio: Option<String>,
    pub mattina_fine: Option<String>,
    pub pomeriggio_inizio: Option<String>,
    pub pomeriggio_fine: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateOrarioCentroInput {
    pub giorno: i64,
    pub attivo: bool,
    pub mattina_inizio: Option<String>,
    pub mattina_fine: Option<String>,
    pub pomeriggio_inizio: Option<String>,
    pub pomeriggio_fine: Option<String>,
}
